Remove backticks from `autoClosingPairs` and `surroundingPairs` to prevent doubling them up

TomasHubelbauer opened the pull request:
Doubling up backticks happens when trying to write a code block in MDX.

This breaks backtick auto-pairing even when writing inline code span, but arguably, it is much less annoying to have to add the matching backtick manually in the case of an inline code span than it is to have to remove it twice every time a code block is being written (in the opening and closing triple-backtick lines).

If you know how to give this file more context so it could only be disabled for code blocks, let me know. Althoug I am not sure it is super desirable to have backtick completions even for inline code spans. But that's just my preference. The original issue reported is facing issues with auto-pairing in inline code spans so I think it is worth just removing this for backticks altogether.

I have added a bit of documentation on how to run the extension. I've written a few VS Code extensions myself but it has been a long time and I've forgotten everything so I had to figure out the instructions. Seems like a waste to not contribute them if there's interest in taking them in.

Initially I thought the way to fix this would be to spy on the LSP messages and find the one responsible for handling the backtick and completing the matching pair, hence why the instructions contain information about LSP messages. In the end it was a much more low level feature so I did not find anything useful in the LSP messages.

Resolves #231
Closes GH-231

BTW would be cool to link to these in the README:
- https://mdxjs.com/community/contribute
- https://mdxjs.com/community/support

I never look for contributions guidelines on the project's website of all places. I am just now noticing this due to the issue template that was applied to my PR. I expect stuff like that to be in the repository. Maybe it is just me but seems like there would be no harm in having it in both places at least.

remcohaszing commented on README.md:
```diff
@@ -85,6 +85,10 @@ See [§ Sponsor][sponsor] on our site for how to help financially.
 </tr>
 </table>
 
+## Contributing
+
+See the `CONTRIBUTING.md` in the `packages` subdirectories that have it.
+
```
None of the packages have a contributing document.

```suggestion
```

TomasHubelbauer replied:
I added one for the VS Code extension that's why I added this. Should I remove it?

remcohaszing replied:
Did you mean you added `.github/README.md`? Anyway, yes, please remove it. :)

TomasHubelbauer replied:
Yeah - but just removed like you said.

remcohaszing commented on packages/vscode-mdx/.github/README.md:
```diff
@@ -0,0 +1,15 @@
+  - Place and hit breakpoints at desired code sites - the debugger will break
+    (This extends to files in `language-server` as well.)
+- Use the Output channel named Markdown Language Server in the main VS Code
+  window (not the Extension Host window) to monitor LSP communication
```
The contributing guidelines are inferred from the [`mdx-js/.github`](https://github.com/mdx-js/.github) repo. This links to https://mdxjs.com/community/contribute/, which contains instructions specifically for the [`mdx-js/mdx`](https://github.com/mdx-js/mdx) repo. I agree this is not ideal.

I suppose there are two things to do here:

1. Have the organization contributing guidelines point to something more generic (we have more repositories). @mdx-js/members thoughts?
2. Write specific contributing guidelines for this project. This project has some specific instructions that don’t apply to other projects, like how to debug a VSCode extension, or how to run the Monaco editor demo.

---

This document is not related to the changes in this PR though. I also don’t think we’ll end up with a file named `.github/README.md`. Could you remove it?

TomasHubelbauer replied:
I think #2 is better but that's from my perspective as a drive by contributor so it might not be the best choice for the org. My angle here is that when a user of and of the packages has an issue and wants to contribute, familiarizing with the structure of the whole org and stuff is friction. Of course that may be my laziness speaking but most of the time when I contribute to tools I use it is to fix a pain point of my and only if I like the process and feedback think about the org and whether I want to be involved more holistically. Again not sure if this is wise to encourage this but that's where I'm at. 

BTW I created the `.github/README.md` over the normal README because that is the extension marketplace README right? I didn't think build instructions were a fit for that. The dot github readme (sorry I'm on my phone) is a feature of Github which allows replacing what readme renders for the directory which seemed appropriate. Do you want me to move the contribution instructions to the real readme or yeet them altogether?

remcohaszing replied:
Options 1 and 2 don’t exclude each other. I think we should to both.

`README.md` will indeed be rendered in the extension marketplace, so it’s good to keep this somewhere else. Typically something like this exists in a file named `CONTRIBUTING.md`.

TomasHubelbauer replied:
Ah okay, I see what you mean. Let me just remove all of the README changes I made, you have a better handle on how to document the contributing guidelines.

remcohaszing reviewed (changes requested):
I’m personally not a fan of auto closing tags anyway :shrug:

I can imagine even people who do use it, prefer not having it for backticks in markdown/MDX over false positives. So I’m fine with removing it.

Could you add a changeset? That’s a markdown file in the `.changesets` directory that looks something like this:

```md
---
vscode-mdx: patch
---

Remove backticks from `autoClosingPairs` and `surroundingPairs`.
```

TomasHubelbauer commented:
Will add a changeset today 👌

TomasHubelbauer commented:
@remcohaszing I added the changeset using `npx changeset` and it gave the changeset file a super weird name `three-cars-happen.md` so I renamed it, hopefully the name doesn't matter and I didn't break anything by doing it. Other than that it should be fine. I wasn't super duper sure if I should place it in the repository directory `.changeset` or create that folder in the package folder but the tool placed it at the root so I assume that's correct.

remcohaszing commented:
The changeset name doesn’t really matter. I prefer something descriptive, like you used. :)

TomasHubelbauer commented:
@remcohaszing Addressed all your feedback. Thank you for the quick review, I appreciate it!

remcohaszing commented:
Looks good! I’ll release it this week. Thank you!
